### PR TITLE
Add setters to RGBA & missing builder methods

### LINKED ISSUE #713: gdk rgba no longer implement DerfMut
We can no longer modify a gdk::RGBA's values & there are no setters on it neither. One or both should be provided

### PR DESCRIPTION
**Pull request by bilelmoussaoui**
Fixes #713

### CODE REVIEW

File: gdk4/src/rgba.rs [hunk 1]
```diff
@@ -71,22 +71,46 @@ impl RGBA {
...
+    /// This method returns an instance of [`RGBABuilder`] which can be used to create [`RGBA`] objects.
+    pub fn builder() -> RGBABuilder {
+        RGBABuilder::default()
+    }
+
     pub fn red(&self) -> f32 {
         self.0.red
     }
 
+    pub fn set_red(&self, red: f32) {
```
Comment (sdroege) [inline gdk4/src/rgba.rs:86]:
```suggestion
    pub fn set_red(&mut self, red: f32) {
```

Also for the others :scream:

File: gdk4/src/rgba.rs [hunk 2]
```diff
@@ -154,3 +178,16 @@ impl FromStr for RGBA {
         RGBA::parse(s)
     }
 }
+
+impl std::ops::Deref for RGBA {
```
Comment (sdroege) [inline gdk4/src/rgba.rs:182]: Why?

Comment (bilelmoussaoui) [reply gdk4/src/rgba.rs:182]: I think setters only are fine, but I had code where I get the RGBA from the gtk::StyleContext & just change the alpha value of the color; a setter should be fine I guess :)

**Review by sdroege — APPROVED** [review]
You copy&paste that to gtk3-rs?

**Comment by bilelmoussaoui** [comment]
> You copy&paste that to gtk3-rs?

https://github.com/gtk-rs/gtk3-rs/pull/673

**Pull request merged**